catalog/dags/data_refresh/alter_data.py

```diff
@@ -0,0 +1,209 @@
+"""
+# Alter Data Task
+
+This module contains the Airflow tasks used for filtering, transforming, and augmenting
+data from the newly created temporary table in the downstream (API) database.
+Currently, this provides an entrypoint for filtering tags that do not meet a certain
+minimum accuracy, are malformed, or are from unvetted providers.
+"""
+
+import logging
+from textwrap import dedent
+
+from airflow.decorators import task, task_group
+from airflow.models.abstractoperator import AbstractOperator
+from airflow.providers.common.sql.hooks.sql import fetch_all_handler, fetch_one_handler
+from psycopg2.extras import Json
+
+from common.constants import POSTGRES_API_CONN_IDS, Environment
+from common.sql import PGExecuteQueryOperator, PostgresHook
+from data_refresh.data_refresh_types import DataRefreshConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_BATCH_SIZE = 100_000
+# Filter out tags that exactly match these terms. All terms should be lowercase.
+TAG_DENYLIST = {
+    "no person",
+    "squareformat",
+    "uploaded:by=flickrmobile",
+    "uploaded:by=instagram",
+    "flickriosapp:filter=flamingo",
+    "by",
+}
+
+# Filter out tags that contain the following terms. All entries should be
+# lowercase.
+TAG_CONTAINS_DENYLIST = {
+    "flickriosapp",
+    "uploaded",
+    ":",
+    "=",
+    "cc0",
+    "by-nc",
+    "by-nd",
+    "by-sa",
+    "by-nc-nd",
+    "by-nc-sa",
+    "pdm",
+}
+
+# Filter out low-confidence tags, which indicate that the machine-generated tag
+# may be inaccurate.
+TAG_MIN_CONFIDENCE = 0.90
+# Filter out tags that match the following providers (either because they haven't
+# been vetted or because they are known to be low-quality).
+FILTERED_TAG_PROVIDERS = {"rekognition"}
+# Select the data within a given batch
```

Comment:
Do you think there'd be any utility in having these come from Airflow variables? My feeling is that if we wanted to adjust these it would probably be worth a PR and approval process anyway, but I thought I'd throw the question out there. Any reason we might want to be able to filter out different tag providers more quickly, for example?

Reply:
Since there's _so much_ heavy lifting with the data refresh process (and probably as you mention, the act of having to make a PR for the change will serve as documentation in some form for the change itself), I think it's best not to have this be configured by a Variable. It's a good thing to be thinking about! And if we find we're changing this sort of thing frequently (_after_ the Rekognition project, at least haha), then we can add that functionality 😊